purchase_propagate_qty/models/purchase.py

```diff
@@ -20,11 +20,46 @@ def _propagage_qty_to_moves(self):
         for line in self:
             if line.state != "purchase":
                 continue
-            for move in line.move_dest_ids | line.move_ids:
-                if move.state in ("cancel", "done"):
-                    continue
-                if move.product_id != line.product_id:
-                    continue
-                move.product_uom_qty = line.product_uom_qty
-                if float_is_zero(move.product_uom_qty, line.product_uom.rounding):
-                    move._action_cancel()
+            moves = line.move_dest_ids | line.move_ids
```

Comment:
I don't like this fix... it seems overcomplicated. If I understand well, it has been done to handle a situation where we increase/decrease qty of a purchase order line having several moves.
The question is: in which situation we have several available moves related to a purchase order line? Isn't the issue is to have several moves here?

Reply:
still, if it can happen, then we have to fix this.
I'll investigate further.

Reply:
I'll create a new pr for that, let's discuss about this on it.

Reply:
Link to the new PR: https://github.com/OCA/purchase-workflow/pull/1233